Content.Server/Guardian/GuardianSystem.cs

```diff
@@ -0,0 +1,355 @@
+            // Use map position so it's not inadvertantly parented to the host + if it's in a container it spawns outside I guess.
+            var guardian = EntityManager.SpawnEntity(comp.GuardianProto, hostXform.MapPosition);
+
+            host.GuardianContainer.Insert(guardian);
+            host.HostedGuardian = guardian;
+
+            if (TryComp(guardian, out GuardianComponent? guardianComponent))
+            {
+                guardianComponent.Host = ev.Target;
+
+                // Grant the user the recall action and notify them
+                actions.Grant(ActionType.ManifestGuardian);
+                SoundSystem.Play(Filter.Entities(ev.Target), "/Audio/Effects/guardian_inject.ogg", ev.Target);
+
+                _popupSystem.PopupEntity(Loc.GetString("guardian-created"), ev.Target, Filter.Entities(ev.Target));
+                // Exhaust the activator
+                comp.Used = true;
+            }
+            else
+            {
+                Logger.ErrorS("guardian", $"Tried to spawn a guardian that doesn't have {nameof(GuardianComponent)}");
+                EntityManager.QueueDeleteEntity(guardian);
+            }
+        }
+
+        /// <summary>
+        /// Triggers when the host receives damage which puts the host in either critical or killed state
+        /// </summary>
+        private void OnHostStateChange(EntityUid uid, GuardianHostComponent component, MobStateChangedEvent args)
+        {
+            if (!component.HostedGuardian.IsValid()) return;
+
+            if (args.CurrentMobState.IsCritical())
+            {
+                _popupSystem.PopupEntity(Loc.GetString("guardian-critical-warn"), component.HostedGuardian, Filter.Entities(component.HostedGuardian));
+                SoundSystem.Play(Filter.Entities(component.HostedGuardian), "/Audio/Effects/guardian_warn.ogg", component.HostedGuardian);
+            }
+            else if (args.CurrentMobState.IsDead())
+            {
+                SoundSystem.Play(Filter.Pvs(uid), "/Audio/Voice/Human/malescream_guardian.ogg", uid, AudioHelpers.WithVariation(0.20f));
+                EntityManager.RemoveComponent<GuardianHostComponent>(uid);
+            }
+        }
+
+        /// <summary>
+        /// Handles guardian receiving damage and splitting it with the host according to his defence percent
+        /// </summary>
+        private void OnGuardianDamaged(EntityUid uid, GuardianComponent component, DamageChangedEvent args)
+        {
+            if (args.DamageDelta == null ||
+                !HasComp<DamageableComponent>(uid) ||
+                !TryComp<DamageableComponent>(component.Host, out var hostDamage)) return;
+
+            _damageSystem.SetDamage(hostDamage, (hostDamage.Damage + args.DamageDelta * component.DamageShare));
+            _popupSystem.PopupEntity(Loc.GetString("guardian-entity-taking-damage"), component.Host, Filter.Entities(component.Host));
```

Comment:
So couple of things:

Firstly, if ever the guardian is susceptible to damage types that the host is not, then this SetDamage call here will make the host susceptible. E.g., if the host is immune to `cold`, but the guardian is not, then after this call the host will now be susceptible to cold damage because their damage dictionary now has a cold entry. There's an `ExclusiveAdd` function that avoids that.

Secondly, this here does not distinguish between healing and damage, so any healing applied to the guardian is also applied to the host. Is that intentional? If it is, you still need to be careful here to not set the hosts health below zero.
Either clamp the damage change, or clamp the final damage that you set it to.

So maybe this should be changed to something like:
```suggestion
            var damageToAdd = args.DamageDelta * component.DamageShare;
            damageToAdd.ClampMin(FixedPoint2.Zero);

            DamageSpecifier newDamage = new(hostDamage.Damage);
            newDamage.ExclusiveAdd(damageToAdd);

            _damageSystem.SetDamage(hostDamage, newDamage);
            _popupSystem.PopupEntity(Loc.GetString("guardian-entity-taking-damage"), component.Host, Filter.Entities(component.Host));
```

Reply:
Actually, why not just use `TryChangeDamage` directly with ignored resistances?
Though you might still want to clamp the damage-change sent to `TryChangeDamage`.

Reply:
I have no idea how it works on /trg/ so you'd have to ask crude. I've just changed it to trychangedamage for now.